homeassistant/components/vesync/diagnostics.py

```diff
@@ -0,0 +1,141 @@
+"""Diagnostics support for VeSync."""
+from __future__ import annotations
+
+from typing import Any
+
+from pyvesync import VeSync
+
+from homeassistant.components.diagnostics import REDACTED
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.device_registry import DeviceEntry
+
+from .common import VeSyncBaseDevice
+from .const import DOMAIN, VS_MANAGER
+
+KEYS_TO_REDACT = {"manager"}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    manager: VeSync = hass.data[DOMAIN][VS_MANAGER]
+
+    data = {}
+    data.update(
+        {
+            DOMAIN: {
+                "account_id": manager.account_id,
+                "bulb_count": len(manager.bulbs),
+                "fan_count": len(manager.fans),
+                "outlets_count": len(manager.outlets),
+                "switch_count": len(manager.switches),
+                "timezone": manager.time_zone,
+            }
+        }
+    )
+
+    data.update(
+        {
+            "home_assistant": {
+                "disabled_by": entry.disabled_by,
+                "disabled_polling": entry.pref_disable_polling,
+            }
+        }
+    )
+
+    data.update(
+        {
+            "devices": {
+                "bulbs": [_redact_device_values(device) for device in manager.bulbs],
```

Comment:
I would expect all info of device diagnostics to be inside a config entry diagnostics. But that doesn't look to be the same ?

Reply:
The devices here are coming directly from the VeSync manager.  I idid this so that a user would be able to pull the config entry diagnostics and find out what the VeSync API is reporting.  The device diagnostics report on the actual Home Assistant devices.

I did this because up until recently, the vesync configuration would skip some devices reported by the API.  The diagnostics will allow users to determine what is being sent by the API and be able to create issues to improve the home assistant device discovery.